tests/github-plugin-diff.js

```diff
@@ -61,209 +60,188 @@ github.authenticate({
-  console.log('No changes in plugin or fixture files since last commit.');
-  process.exit(0);
-})
-.then(() => {
-  let message = [];
-  message.push(`${identification}`);
-  message.push(`<!-- commit = ${process.env.TRAVIS_COMMIT} -->`);
-  message.push('Last updated: ' + new Date(Date.now()).toLocaleString());
-  message.push('## Diff plugin outputs test');
+// generate diff tasks describing the diffed plugins, fixtures and the reason for diffing (which component has changed)
+.then(changedComponents => {
+  return new Promise((resolve, reject) => {
+    const allPlugins = exportPlugins.filter(plugin => !changedComponents.added.plugins.includes(plugin));
+    const allTestFixtures = testFixtures.filter(fixture => !changedComponents.added.fixtures.includes(fixture));
 
-  if (Object.keys(fixtureData).length === 0 && Object.keys(pluginData).length === 0) {
-    message.push('*No fixture or plugin files were changed in this PR.*');
-  }
-  else {
-    for (let fixture of Object.keys(fixtureData)) {
-      message.push(`### Modified \`${fixture}\` in this PR`);
-
-      let hasContent = false;
-      for (let plugin of Object.keys(fixtureData[fixture])) {
-        const pluginMessage = printPlugin(fixtureData[fixture][plugin]);
-        if (pluginMessage.length > 0) {
-          hasContent = true;
-          message.push(`#### Output changed for plugin \`${plugin}\``);
-          message = message.concat(pluginMessage);
-        }
+    if (changedComponents.modified.model) {
+      diffTasks.push({
+        type: 'model',
+        plugins: allPlugins,
+        fixtures: allTestFixtures
+      });
+    }
+    else {
+      for (const plugin of changedComponents.modified.plugins) {
+        diffTasks.push({
+          type: 'plugin',
+          plugins: [plugin],
+          fixtures: allTestFixtures
+        });
       }
+    }
 
-      if (!hasContent) {
-        message.push('Output files not changed.');
-      }
+    for (const fixture of changedComponents.modified.fixtures) {
+      diffTasks.push({
+        type: 'fixture',
+        plugins: allPlugins,
+        fixtures: [fixture]
+      });
     }
 
-    for (let plugin of Object.keys(pluginData)) {
-      message.push(`### Modified plugin \`${plugin}\` in this PR`);
-      message.push('Plugins are always tested with the following fixtures:');
-      for (const testFixture of testFixtures) {
-        message.push(` - ${testFixture}`);
-      }
+    resolve();
+  });
+})
+// run the diff tasks (if there are some)
+.then(() => {
+  if (diffTasks.length === 0) {
+    console.log('Model, plugins and fixtures not modified.');
+    process.exit(0);
+  }
 
-      const pluginMessage = printPlugin(pluginData[plugin]);
-      if (pluginMessage.length > 0) {
-        message = message.concat(pluginMessage);
-      }
-      else {
-        message.push('Output files not changed.');
-      }
-    }
+  return Promise.all(diffTasks.map(task => {
+    return new Promise((resolve, reject) => {
```

Comment:
`Promise.all(...).then(...)` returns itself (a promise), so you don't need to wrap it in another one